python/ray/util/metrics.py

```diff
@@ -14,6 +15,25 @@
 
 logger = logging.getLogger(__name__)
 
+METRIC_NAME_RE = re.compile(r"^[a-zA-Z_:][a-zA-Z0-9_:]*$")
+"""
+    Validation logic is copied from prometheus_client.metrics_core
+    https://github.com/prometheus/client_python/blob/2dcd17efd0ce2f0a1ad15cb3c150ffcdc42ced65/prometheus_client/metrics_core.py#L11
+"""
+
+
+def _validate_metric_name(name: str):
+
+    if len(name) == 0:
+        raise ValueError("Empty name is not allowed. Please provide a metric name.")
```

Comment:
[anyscalesam](https://github.com/anyscalesam) doc test are failing. Not sure why since no changes are done there. Can you please help here?